graphix/gflow.py

```diff
@@ -87,145 +29,142 @@ def gflow(g, v_in, v_out, meas_plane=None, timeout=1000):
-            inds = np.where(Ix)[0]
-
-            if len(inds) > 0:  # has solution
-                c_set = c_set | {u}
-                g[u] = set(v_correct_list[ind_] for ind_ in inds)
-                l_k[u] = k
-    if not c_set:
-        finished = True
-        if v_out == v:
-            exist = True
+    nodes = set(graph.nodes)
+    if output == nodes:
+        return g, l_k
+    non_output = nodes - output
+    correction_candidate = output - input
+    solver = GF2Solver()
+    adjacency_matrix, node_order_list = get_adjacency_matrix(graph)
+    adjacency_matrix_row_reduced = remove_nodes_from_row(adjacency_matrix, node_order_list, output)
+    adjacency_matrix_reduced = remove_nodes_from_column(
+        adjacency_matrix_row_reduced, node_order_list, nodes - correction_candidate
+    )
+
+    node_order_row = []
+    node_order_col = []
+    for node in node_order_list:
+        if node in non_output:
+            node_order_row.append(node)
+        if node in correction_candidate:
+            node_order_col.append(node)
+
+    RHS_map = dict()
+    for node in non_output:
+        vec = np.zeros(len(node_order_row))
+        if meas_planes[node] == "XY":
+            vec[node_order_row.index(node)] = 1
+        elif meas_planes[node] == "XZ":
+            vec[node_order_row.index(node)] = 1
+            vec_add = adjacency_matrix_row_reduced[:, node_order_row.index(node)].reshape(vec.shape)
+            vec = (vec + vec_add) % 2
+        elif meas_planes[node] == "YZ":
+            vec = adjacency_matrix_row_reduced[:, node_order_row.index(node)].reshape(vec.shape)
+        RHS_map[node] = vec
+
+    all_solutions, uncertainty = solver.solve(adjacency_matrix_reduced, RHS_map, node_order_col)
+
+    corrected = set()
+    for candidate in all_solutions.keys():
+        if len(all_solutions[candidate]) == 0:
+            continue
+        if meas_planes[candidate] in ["XZ", "YZ"]:
+            all_solutions[candidate][candidate] = SolutionNode(candidate, 1, set())
+        if mode == "single":
+            g[candidate] = find_single_solution(all_solutions[candidate], uncertainty)
+            l_k[candidate] = k
+        elif mode == "all":
+            g[candidate] = find_all_solutions(all_solutions[candidate], uncertainty)
+            l_k[candidate] = k
+        elif mode == "abstract":
+            g[candidate] = all_solutions[candidate]
+            l_k[candidate] = {"layer": k, "uncertainty": uncertainty}
+
+        corrected |= {candidate}
+
+    if len(corrected) == 0:
+        if output == nodes:
+            return g, l_k
         else:
-            exist = False
+            return None, None
     else:
-        finished = False
-        exist = True
+        return gflowaux(graph, input, output | corrected, meas_planes, k + 1, l_k, g, mode=mode)
 
-    return v_out | c_set, g, l_k, finished, exist
 
-
-def flow(g, v_in, v_out, meas_plane=None, timeout=10000):
+def flow(graph, input, output, meas_planes=None):
```

Comment:
dont we need timeout anymore?

Reply:
I don't understand why `timeout` argument is necessary. A maximum iteration time is a number of layers, which is not so large for moderate graphs.

Reply:
I mean, I do get why you removed it, flow finding algorithm is fast enough to be operationally fine without timeout. 
be sure to mention API change in changeling and why it's fine